### PR TITLE
Delegate route prefix change

### PR DESCRIPTION
**Pull request by ilackarms**
currently, routes who delegate have their prefix automatically prepended to the child routes

so the delegation chain

```
/a -> /b -> /c
```

will produce the route 
```
/a/b/c
```

this change *requires* the child route to inherit the prefix from the parent.

so 
 ```
/a -> /b -> /c
```
will no longer be valid. instead the chain must read

```
/a -> /a/b -> /a/b/c
```

BOT NOTES: 
resolves https://github.com/solo-io/gloo/pull/1262

### CODE REVIEW

**Comment by ilackarms** [comment]
this pr also handles correct merging behavior for routePlugins

File: projects/gateway/pkg/translator/http.go [hunk 1]
```diff
@@ -379,3 +400,13 @@ func (rv *routeVisitor) convertDelegateAction(sourceResource resources.InputReso
 
 	return delegatedRoutes, nil
 }
+
+func mergeRoutePlugins(dst, src *gloov1.RoutePlugins) (*gloov1.RoutePlugins, error) {
+	if src == nil {
+		return dst, nil
+	}
+	if dst != nil {
+		return dst, mergo.Merge(dst, *src)
```
Comment (marcogschmidt) [inline projects/gateway/pkg/translator/http.go]: The [`mergo.Merge` function](https://github.com/imdario/mergo/blob/9316a62528ac99aaecb4e47eadd6dc8aa6533d58/merge.go#L198:6) will recursively fill any "empty" field in `dest` if the corresponding field is set in `source` (zero-values are considered "empty", so an `int` field explicitly set to `0` will be overwritten). I am a bit worried that this merge strategy might lead to invalid plugin configurations. Think of a plugin where setting an optional field to an unexpected value might lead to errors.

I think this feature as-is delivers value (I can imagine a common use case being having a plugin config on the top level route and none on the route table -> delegate routes just inherit) so I am OK with it getting into `master`, but we have to **clearly document the merging behavior on the proto**, for users who might have more complex configurations.

Thinking of a more complete approach (that would be well beyond the scope of this PR) I noticed that you can pass [`Transformers`](https://github.com/imdario/mergo/blob/9316a62528ac99aaecb4e47eadd6dc8aa6533d58/merge.go#L33) to the `Merge` function: this allows you to define custom merge semantics for specific types. In the future, we could use this feature to differently handle each of the plugin types that might be problematic (by having them provide their own `Transformation` instance).

Comment (ilackarms) [reply projects/gateway/pkg/translator/http.go]: you're right that the recursive nature of `mergo.Merge` could lead to bad config. what i want to do is only merge the field itself.

maybe I can use the transformer to do that

Comment (marcogschmidt) [reply projects/gateway/pkg/translator/http.go]: I like the shallow merge you added. Can you document the behavior on the gateway `Route.RoutePlugins` proto field? After that I am ok with approving.

Comment (marcogschmidt) [reply projects/gateway/pkg/translator/http.go]: Nevermind, just noticed that you had already added a comment.

**Pull request merged**